### PR TITLE
WIP: Add an Online domain module

### PR DESCRIPTION
**Pull request by remyleone**
##### SUMMARY

Add an Online domain module

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

- online_domain_record

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (online_domain 1b057c6d9f) last updated 2018/09/03 14:59:21 (GMT +200)
  config file = None
  configured module search path = [u'/Users/sieben/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sieben/workspace/ansible/lib/ansible
  executable location = /Users/sieben/workspace/ansible/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

### CODE REVIEW

**Comment by remyleone** [comment]
WIP. DO NOT REVIEW NOW

File: vars_example [hunk 1]
```diff
@@ -0,0 +1,8 @@
+all:
```
Comment (mattclay) [inline vars_example]: This file shouldn't be in the root of the repo, if it was even intended to be included at all.

**Comment by remyleone** [comment]
@mattclay it is a WIP don't lose too much time reviewing it.

File: lib/ansible/modules/cloud/online/online_domain_record.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+```
+ansible-playbook --connection=local -e 'ansible_python_interpreter=/usr/bin/python3' -i vars_example play_testxxxx.yml
+```
+
+## Records
+
+Ensure A record
+```yaml
+- domain_online_record:
```
Comment (broferek) [inline lib/ansible/modules/cloud/online/online_domain_record.py]: The module is misspelled. It should be online_domain_record

**Comment by broferek** [comment]
Hi Remy,

I tried to use it with the following params 

    - online_domain_record:
        token: 'xxxxxx'
        zone: 'yyy-xxxx.com'
        name: 'cname-01'
        type: CNAME
        content: 'sd-XXXXX.dedibox.fr'

The issue is that it has updated all my records with the suffix of the zone 'yyy-xxxx.com'

I'm not sure if it is an expected behaviour of the API or a real module issue.

**Comment by remyleone** [comment]
@broferek WIP. DO NOT REVIEW NOW :) I'm not done with this module

**Pull request closed without merging**